### PR TITLE
add computer name parameter to win_dns_zone module

### PR DESCRIPTION
**Pull request by cjpit**
##### SUMMARY
Adds computer name parameter, similar to win_dns_record

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
win_dns_zone

##### ADDITIONAL INFORMATION
https://docs.microsoft.com/en-us/powershell/module/dnsserver/add-dnsserverprimaryzone?view=win10-ps

### CODE REVIEW

File: plugins/modules/win_dns_zone.py [hunk 1]
```diff
@@ -80,6 +80,13 @@
       - At least one server is required.
     elements: str
     type: list
+  computer_name:
```
Comment (jborean93) [inline plugins/modules/win_dns_zone.py:83]: I'm torn as to whether this should be `server` or `computer_name`. I know the underlying cmdlets use `-ComputerName` but we've already set a precedent for `server` with the AD modules to explicitly select the AD host to target when managing the object.

I personally lean towards making this `server` to make it uniform with what we already do but interested to hear what you think.

File: plugins/modules/win_dns_zone.py [hunk 2]
```diff
@@ -80,6 +80,13 @@
...
+    description:
+      - Specifies a DNS server.
+      - You can specify an IP address or any value that resolves to an IP
+        address, such as a fully qualified domain name (FQDN), host name, or
+        NETBIOS name.
+    type: str
```
Comment (jborean93) [inline plugins/modules/win_dns_zone.py:89]: This is a new option, it should have `version_added: 1.2.0` to tell users when this was first available in the docs.

**Review by jborean93 — CHANGES_REQUESTED** [review]
Changes look good, just wondering whether we should change the option to `server` to match the way things are done in the AD modules. 

The only other thing we want to add is a changelog fragment so we can advertise this new feature when we release the next version of this collection. To add a fragment you need to create a new file that ends with `.yml` in https://github.com/ansible-collections/community.windows/tree/main/changelogs/fragments.

The contents would be something like (feel free to change the text):

```yaml
minor_changes:
- win_dns_zone - Added ``module option`` to specify the DNS server to connect to when managing the zone object
```

File: plugins/modules/win_dns_zone.py (hunk 1 above)
Comment (cjpit) [reply plugins/modules/win_dns_zone.py:83]: Yeah I'd agree, server is easier to understand - When I first read the docs for the module, i didn't get it and it wasn't until i got the powershell cmd working I realized what to do, so yep can make those changes, as well as to win_dns_record

File: plugins/modules/win_dns_record.py [hunk 3]
```diff
@@ -77,12 +77,13 @@
...
     required: yes
     type: str
-  computer_name:
+  server:
     description:
       - Specifies a DNS server.
       - You can specify an IP address or any value that resolves to an IP
         address, such as a fully qualified domain name (FQDN), host name, or
         NETBIOS name.
+    version_added: 1.2.0
```
Comment (jborean93) [inline plugins/modules/win_dns_record.py:86]: Sorry I was not aware that `computer_name` was already in use here, we still need to accept as an alias that for backwards compatibility. We can skip the `version_added` as technically the option isn't new, just make sure the description mentions it was renamed in the 1.2.0 release like so

```suggestion
  server:
    description:
      - Specifies a DNS server.
      - You can specify an IP address or any value that resolves to an IP
        address, such as a fully qualified domain name (FQDN), host name, or
        NETBIOS name.
      - Was renamed from I(computer_name) to I(server) in 1.2.0.
    aliases:
    - computer_name
```

File: plugins/modules/win_dns_record.ps1 [hunk 4]
```diff
@@ -16,7 +16,7 @@ $spec = @{
         value = @{ type = "list"; elements = "str"; default = @() ; aliases=@( 'values' )}
         weight = @{ type = "int"}
         zone = @{ type = "str"; required = $true }
-        computer_name = @{ type = "str" }
+        server = @{ type = "str" }
```
Comment (jborean93) [inline plugins/modules/win_dns_record.ps1:19]: Need to make sure we don't break any existing playbooks by accepting it as an alias

```suggestion
        server = @{ type = "str"; aliases = @("computer_name") }
```

**Pull request closed without merging**